tools/topology/topology2/doc/CMakeLists.txt

```
@@ -0,0 +1,33 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+cmake_minimum_required(VERSION 3.13)
+
+project(SOF_TOPOLOGY2_DOC NONE)
+
+set(SOF_ROOT_SOURCE_DIRECTORY "${PROJECT_SOURCE_DIR}/..")
+
+set(top_srcdir "${SOF_ROOT_SOURCE_DIRECTORY}")
+
+configure_file(
+	"${PROJECT_SOURCE_DIR}/sof.doxygen.in"
+	"${PROJECT_BINARY_DIR}/sof.doxygen"
+)
+
+add_custom_command(
+	OUTPUT ${PROJECT_SOURCE_DIR}/extra-contents/contents.doxy
```

```suggestion
	OUTPUT ${PROJECT_BINARY_DIR}/extra-contents/contents.doxy
```
Never create files outside the build directory, this is confusing both users and CMake.

Please also use `set()` to define a variable and avoid duplication.